sqtom/tests/test_fitting_1d.py

```diff
@@ -15,8 +15,45 @@
 
 import pytest
 import numpy as np
+from thewalrus.samples import photon_number_sampler
 from sqtom.forward_solver import degenerate_pmf
-from sqtom.fitting_1d import fit_1d, threshold_1d, marginal_calcs_1d
+from sqtom.fitting_1d import (
+    two_schmidt_mode_guess,
+    marginal_calcs_1d,
+    gen_hist_1d,
+    threshold_1d,
+    fit_1d,
+)
+
+
+@pytest.mark.parametrize("sq_0", [0.1, 1.0, 2.0])
+@pytest.mark.parametrize("eta", [0.1, 0.5, 1.0])
+def test_gen_hist_1d(sq_0, eta):
+    """Check that a histogram is constructed correctly for a degenerate squeezing source"""
+    nsamples = 1_000_000
+    nmax = 100
+    pmf_init = degenerate_pmf({"sq_0": sq_0, "n_modes": 1, "eta": eta}, cutoff=nmax)
+    samples = photon_number_sampler(pmf_init, nsamples)
+    pmf_gen = gen_hist_1d(samples)
+    pmf_final = degenerate_pmf({"sq_0": sq_0, "n_modes": 1, "eta": eta}, cutoff=np.max(samples))
+    assert np.allclose(pmf_gen, pmf_final, atol=0.02)
+
+
+@pytest.mark.parametrize("eta", [0.1, 0.5, 1.0])
+@pytest.mark.parametrize("sq_0", [0.1, 1.0, 2.0])
+@pytest.mark.parametrize("sq_1", [0.0, 0.1, 1.0])
+def test_two_schmidt_mode_guess_exact(eta, sq_0, sq_1):
+    """Test that one can invert correctly when there are two Schmidt modes
+    and no dark counts.
+    """
+    pmf = degenerate_pmf({"sq_0": sq_0, "sq_1": sq_1, "n_modes": 2, "eta": eta})
+    guess = two_schmidt_mode_guess(pmf)
+    assert np.allclose(eta, guess["eta"], atol=1.0e-2)
+    sq_ns = [sq_0, sq_1]  # We need to sort sq_n1 and sq_n2 so that sq_n1 >= sq_n2
```

Comment:
(Minor suggestion): further to the suggestion, might be worth leaving another note on why the sorting is required (perhaps was just unclear for me though :slightly_smiling_face: )
```suggestion
    sq_ns = [sq_0, sq_1]  # We need to sort sq_n0 and sq_n1 so that sq_n0 >= sq_n1
```

Reply:
The reason for the sorting is that `two_schmidt_mode_guess` always returns a guess with the property that `sq_0 >= sq_1`, and thus to compare you need to sort the values that were given as input.